crates/router/src/connector/trustpay.rs

```diff
@@ -0,0 +1,662 @@
+            .ok_or_else(|| errors::ConnectorError::MissingConnectorRefundID)?;
+        Ok(format!(
+            "{}{}/{}",
+            self.base_url(connectors),
+            "api/v1/instance",
+            id
+        ))
+    }
+
+    fn build_request(
+        &self,
+        req: &types::RefundSyncRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Option<services::Request>, errors::ConnectorError> {
+        match req.payment_method {
+            storage_models::enums::PaymentMethod::BankRedirect => {
+                Err(errors::ConnectorError::WebhooksNotImplemented).into_report()
+            }
+            _ => Ok(Some(
+                services::RequestBuilder::new()
+                    .method(services::Method::Get)
+                    .url(&types::RefundSyncType::get_url(self, req, connectors)?)
+                    .headers(types::RefundSyncType::get_headers(self, req, connectors)?)
+                    .build(),
+            )),
+        }
+    }
+
+    fn handle_response(
+        &self,
+        data: &types::RefundSyncRouterData,
+        res: Response,
+    ) -> CustomResult<types::RefundSyncRouterData, errors::ConnectorError> {
+        let response: trustpay::RefundResponse = res
+            .response
+            .parse_struct("trustpay RefundResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        types::RouterData::try_from(types::ResponseRouterData {
+            response,
+            data: data.clone(),
+            http_code: res.status_code,
+        })
+        .change_context(errors::ConnectorError::ResponseHandlingFailed)
+    }
+
+    fn get_error_response(
+        &self,
+        res: Response,
+    ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
+        self.build_error_response(res)
+    }
+}
+
+#[async_trait::async_trait]
+impl api::IncomingWebhook for Trustpay {
+    fn get_webhook_object_reference_id(
+        &self,
+        _request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Err(errors::ConnectorError::WebhooksNotImplemented).into_report()
+    }
+
+    fn get_webhook_event_type(
+        &self,
+        _request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<api::IncomingWebhookEvent, errors::ConnectorError> {
+        Err(errors::ConnectorError::WebhooksNotImplemented).into_report()
+    }
+
+    fn get_webhook_resource_object(
+        &self,
+        _request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<serde_json::Value, errors::ConnectorError> {
+        Err(errors::ConnectorError::WebhooksNotImplemented).into_report()
+    }
+}
+
+impl services::ConnectorRedirectResponse for Trustpay {
+    fn get_flow_type(
+        &self,
+        _query_params: &str,
+    ) -> CustomResult<payments::CallConnectorAction, errors::ConnectorError> {
+        Ok(payments::CallConnectorAction::Trigger)
```

Comment:
Can we avoid few psync calls by using the status update handler?

Reply:
Done